common/src/main/java/com/wynntils/services/itemfilter/type/ItemFilter.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.services.itemfilter.type;
+
+import com.google.common.base.CaseFormat;
+import com.wynntils.utils.type.ErrorOr;
+import java.util.Collections;
+import java.util.List;
+import net.minecraft.client.resources.language.I18n;
+
+/**
+ * An {@link ItemFilter} is both : <br>
+ * - A model to represent a filter, holding its name, aliases, and translation keys. <br>
+ * - A factory that will check the validity of an input string for this filter and create an
+ *   {@link ItemFilterMatcher} from it.<br>
+ * <br>
+ * The actual logic of the filter (i.e. the code that will check if an item matches the filter) is
+ * implemented in the {@link ItemFilterMatcher} returned by {@link #createMatcher(String)}.
+ */
+public abstract class ItemFilter {
+    protected final String name;
+    protected final List<String> aliases;
+    protected final String translationName;
+
+    protected ItemFilter(List<String> aliases) {
+        this.aliases = aliases;
+
+        String name = this.getClass().getSimpleName().replace("ItemFilter", "");
+        this.name = CaseFormat.UPPER_CAMEL.to(CaseFormat.LOWER_UNDERSCORE, name);
+        this.translationName = CaseFormat.UPPER_CAMEL.to(CaseFormat.LOWER_CAMEL, name);
+    }
+
+    /**
+     * Creates a new item filter matcher from the given input string.
+     *
+     * @param inputString the input string
+     * @return the created item filter, or a translated error string if the input string is invalid
+     */
+    public abstract ErrorOr<? extends ItemFilterMatcher> createMatcher(String inputString);
+
+    public String getName() {
+        return name;
+    }
+
+    public List<String> getAliases() {
+        return Collections.unmodifiableList(aliases);
+    }
+
+    protected String getTranslationName() {
+        return translationName;
+    }
+
+    protected String getTranslation(String keySuffix, Object... formatValues) {
+        return I18n.get("feature.wynntils.itemFilter." + getTranslationName() + "." + keySuffix, formatValues);
```

Comment:
```suggestion
        return I18n.get("service.wynntils.itemFilter." + getTranslationName() + "." + keySuffix, formatValues);
```
(and also update en_us.json)

Reply:
I also think it would make more sense. However other services using translations use "feature." (`UpdateService` & `LootrunService`). Do you still want to make the change ?

Reply:
Yes. We'll fix the other services separately. (Good catch though, I had not noticed we did this.)